main.py

```diff
@@ -116,29 +116,19 @@ def process(data, factors):
     return average
 
 def choose_direction(data, average):
-    user_surfaces = []
-    for direction_name, surface_arrays in data.items():
-        user_surfaces.append((direction_name, len(surface_arrays.keys())))
-    #get the number of surfaces stored in average as keys
-    user_surfaces = sorted(user_surfaces, key=lambda x: x[1], reverse=True)
-    chosen_surface_num = user_surfaces[0][1]
-    average_surface_num = len(average[next(iter(average))].keys())
-    #I am not sure if this is really necessary to check if they are equal
-    if average_surface_num == chosen_surface_num:
-        return user_surfaces[0][0]
-    
+    data_direction_surfaces = [(direction, data_surfaces.keys()) for direction, data_surfaces in data.items()]
+    average_surfaces = [average_surface for average_surface in average[next(iter(average))].keys()]
+    for direction, data_surfaces in data_direction_surfaces:
+        if all(average_surface in data_surfaces for average_surface in average_surfaces):
```

Comment:
@harrywhoforks I like this line. But what if the statement is False?

Reply:
@harrywhoforks sure, or you can `raise` a custom error back to user. Either way will work.

Reply:
i am not sure how to use raise lol. I pushed anyway. so it is now like if the no True from the if-condition for all data_surfaces in data_direction_surfaces, then jump to next line and print something then quit.

Reply:
@harrywhoforks ```raise``` is a syntax that lets you throw errors and halt operations. You can throw existing errors and of course you can just make one up like ```NoOneDoThatInRealLifeError```.

Reply:
> so it is now like if there is no True from the if-condition for all data_surfaces in data_direction_surfaces, then jump to next line and print something then quit.

Using commands like ```quit()``` is disencouraged and even more absurd when you put it inside a function that ```choose_direction```; a function should do what its name says, and ideally doing just one thing only. I would recommend you that to return a empty string, or ```None``` object and to check if ```choose_direction``` found a valid direction, and act upon in outside the function (aka ```main()```).

Reply:
@harrywhoforks Looking good! Resolving the comment.